lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -783,12 +784,21 @@ def get_thread_list(
     if count_flagged and not context["is_requester_privileged"]:
         raise PermissionDenied("`count_flagged` can only be set by users with moderator access or higher.")
 
+    group_id = (
+        None if context["is_requester_privileged"] else
+        get_group_id_for_user(request.user, CourseDiscussionSettings.get(course.id))
+    )
+
+    if request.GET.get("group_id", None):
+        if Role.user_has_role_for_course(request.user, course_key, ['Administrator', 'Community TA', 'Moderator']):
```

Comment:
You can use these constants here
`FORUM_ROLE_ADMINISTRATOR,
    FORUM_ROLE_COMMUNITY_TA,
    FORUM_ROLE_MODERATOR,`